app/views/components/_banner.html.erb

```diff
@@ -2,30 +2,25 @@
   add_app_component_stylesheet("banner")
 
   title ||= false
-  aside ||= false
-  large_text = !title && !aside
+  secondary_text ||= false
   local_assigns[:margin_bottom] ||= 7
 
   component_helper = GovukPublishingComponents::Presenters::ComponentWrapperHelper.new(local_assigns)
   component_helper.add_class("app-c-banner")
-  component_helper.add_class("app-c-banner--aside") if aside
   component_helper.add_aria_attribute({ label: "Notice" })
   component_helper.set_lang("en")
   component_helper.add_data_attribute({ module: "ga4-link-tracker" })
   component_helper.add_data_attribute({ ga4_track_links_only: "" })
   component_helper.add_data_attribute({ ga4_link: { event_name: "navigation", type: "callout" } })
 %>
-<% content_block = capture do %>
+<%= tag.section(**component_helper.all_attributes) do %>
   <% if title %>
```

Comment:
Just wondering if `title` should be optional 🤔 

At present the `title` attribute appears to always be set for the banner component in the [call_for_evidence](https://github.com/alphagov/government-frontend/blob/8799336fd3c96dac7c7e53cce3cd8a522beb17da/app/views/content_items/call_for_evidence.html.erb#L105) and [consultation](https://github.com/alphagov/government-frontend/blob/8799336fd3c96dac7c7e53cce3cd8a522beb17da/app/views/content_items/consultation.html.erb#L132) view templates.

If it is optional, then we may want to keep one of the examples in the docs where only `text` is provided.

Reply:
Good point, I was going to on the approach that `title` would be always required as that's how the component is being used right now. But happy to make it optional, I'll add an example for it too 🙂 

> this means that when only text is provide to the component, there is more space at the top than expected.

I think this relates to the same point about right now the title being required. But I'll fix the spacing now that we want to make the title optional so that only text can be passed through 👍

Reply:
For anyone else reading, after discussing this IRL, we've decided to actually make the component enforce the current usage more closely by requiring both the title and text attributes to be passed to it. This reflects how the component is currently used. If the component needed to become more flexible and work for more use cases, we should consider moving it to the gem. (We're generally trying to reduce complexity in apps, instead of adding further unnecessary options; there's also been some discussion about how the gem inverse header component could in the future be made more flexible and replace this banner component so we want to make that easier where possible instead of allowing more use cases and potentially bloating the app component further.)